### DISCUSSION TITLE
mouseover image in html-keyboard-response

### DISCUSSION THREAD

**Discussion by joshkenney** [jspsych/jsPsych#946] [Q&A] [answered]
We are designing a new experiment that makes use of mouseover to remove blur from an image. I have this tested and working in a separate index.html file. I can create a stimulus method to return html with the gaussian filter over the image, but mouseover does nothing. Is mouseover disabled in jsPsych html-keyboard-response?

**Comment by vijaymarupudi** [comment, marked as answer]
Not that I know of! Could you post the code you are using for jsPsych integration?

**Reply by joshkenney** [reply]
Sure! Thanks for the reply! Download the attached experiment.
Run index.php
You can see that before jsPsych loads, the blur works fine.
After you click "consent/next", you will see it does not.
[mouseEye-master 2.zip](https://github.com/jspsych/jsPsych/files/5012364/mouseEye-master.2.zip)

**Reply by vijaymarupudi** [reply]
I can't run your experiment because I do not use PHP, but I'm fairly certain this part is the cause of the error. You probably want the code to run when the trial loads, but as it is currently written, the code will run immediately, returning an undefined to the `on_load` property. After the document has loaded (the DOMContentLoaded event), the callback you provided will run, which is probably not what you wanted.

```javascript
on_load: $(document).ready(function(){
  $("p").mouseover(function(){
    $("p").css("background-color", "yellow");
  });
  $("p").mouseout(function(){
    $("p").css("background-color", "lightgray");
  });
}),
```

You probably want something like this

```javascript
on_load: function(){
  $("p").mouseover(function(){
    $("p").css("background-color", "yellow");
  };
  $("p").mouseout(function(){
    $("p").css("background-color", "lightgray");
  });
}
```

**Reply by trig11** [reply]
Hi, I am trying to do something very similar but cannot get your code to work -- would you mind posting what your first_half variable looks like when it is working now? Also, are you planning on saving the mouse tracking data, as well?

**Reply by joshkenney** [reply]
Yes, we are planning to save the tracking data as well. It sounds like you are interested in collaborating! The experiment is an analog for eye tracking / visual search. I will talk to my collaborator, and will connect you to either clone or fork the repo

**Reply by joshkenney** [reply]
@trig11 I just with with my collaborator Danielle, and we are happy to share the code. Perhaps we can schedule a time to meet on zoom so we can catch you up on the development!
you can find this experiment online at https://belieflab.yale.edu/mouseEye

**Reply by joshkenney** [reply]
https://github.com/DaniPratt/mouseEye

**Comment by joshkenney** [comment]
Thanks for catching this Vijay! It is now working as intended.

I would recommend using XAMPP for developing your jsPsych tasks. XAMPP is a full development stack VM and runs server side operations from your localhost.

Important! newer versions will cause styling conflicts with the jsPsych library.
Please download and use this version of xampp:
https://www.apachefriends.org/xampp-files/7.3.18/xampp-osx-7.3.18-0-vm.dmg

let me know if you need a hand getting it installed.

Thanks again!!!

Josh

**Reply by vijaymarupudi** [reply]
Oh I know PHP, I just prefer to use Python instead :) Although XAMPP shouldn't do anything to the styles, that seems like an odd bug.

Glad that it's working now!

**Reply by joshkenney** [reply]
Oh nice! Do you use something like flask to make the database connection?

**Reply by vijaymarupudi** [reply]
Yep, I use flask for server side API! I store my jsPsych files as plain files on the server, since they are not very structured. For more structured data, [`psycopg2`](https://www.psycopg.org/) does the job :)

**Comment by joshkenney** [comment]
Very cool! I am working on software to organize jsPsych data with MongoDB. Will keep you updated!

**Reply by vijaymarupudi** [reply]
Ah yes, MongoDB would be a good choice for such data! Hopefully the tool will be open source :)

**Reply by joshkenney** [reply]
"Think free as in free speech, not free beer"
